Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp

```diff
@@ -575,13 +576,29 @@ LayoutUnit LineLayout::firstLinePhysicalBaseline() const
     }
 
     auto& firstLine = m_inlineContent->lines.first();
+    return physicalBaselineFromWritingMode(firstLine); 
+}
+
+LayoutUnit LineLayout::lastLinePhysicalBaseline() const
+{
+    if (!m_inlineContent || m_inlineContent->lines.isEmpty()) {
+        ASSERT_NOT_REACHED();
+        return { };
+    }
+
+    Line& lastLine = m_inlineContent->lines.last();
+    return physicalBaselineFromWritingMode(lastLine);
+}
+
+LayoutUnit LineLayout::physicalBaselineFromWritingMode(LayoutIntegration::Line& line) const
```

Comment:
physical (visual) values always incorporate the writing mode (hence the name). I'd just call it something like  physicalBaselineForLine.

Reply:
Changed